### PR TITLE
feat: Add algolia and replace local docs search

### PR DESCRIPTION
**Pull request by searchableguy**
I finally got the response from algolia so adding the support here. 

Though, it seems the index isn't up-to-date.

### CODE REVIEW

**Comment by searchableguy** [comment]
Due to change in docusaurus. Need to re-run crawler from dashboard before we can see contextual results here.

**Comment by wonderwomancode** [comment]
Domain verification is disabled for the crawler and we need to put the new subdomain in there. Sent an email to Algolia support.
Screen Shot 2022-09-15 at 5 37 33 PM

**Comment by searchableguy** [comment]
I wonder who applied for it a few months ago (old index) though with gitbook url.

**Comment by wonderwomancode** [comment]
Response email received from Algolia support Tues. 20th asking what domain we wanted added, response to that sent, now just waiting for them to add it or give us access to add it.

File: docusaurus.config.js [hunk 1]
```diff
@@ -16,103 +16,137 @@ const config = {
   favicon: "img/favicon.ico",
   organizationName: "unlock-protocol", // Usually your GitHub org/user name.
   projectName: "docs", // Usually your repo name.
-  plugins: [require.resolve("@cmfcmf/docusaurus-search-local"),
-  [
-    require.resolve('docusaurus-gtm-plugin'),
-    {
-      id: 'GTM-PRCCFV9', // GTM Container ID
-    }
+  plugins: [
```
Comment (julien51) [inline docusaurus.config.js:19]: These changes are annoying. We should try to find a way to use the same format for everyone... linter maybe?

**Review by julien51 — APPROVED** [review]
LGTM

**Pull request merged**